src/pages/ResearcherProfile.js

```diff
@@ -1,492 +1,351 @@
+        generateCountryNames();
+        generateStateNames();
+
+        props.history.push('profile');
+        setNotificationData(await NotificationService.getBannerObjectById('eRACommonsOutage'));
+
+        setCurrentUser(Storage.getCurrentUser());
+        setIsLoading(false);
+      } catch (error) {
+        Notification.showError({text: 'Error: Unable to retrieve user data from server'});
+      } finally {
+        setIsLoading(false);
+      }
     };
-  }
 
-  getResearcherProfile = async () => {
-    const user = await User.getMe();
-    const researcherProps = getPropertyValuesFromUser(user);
-    const institutionList = await Institution.list();
-
-    this.setState(prev => {
-      if (isEmpty(user.roles)) {
-        prev.roles = [{ 'roleId': 5, 'name': USER_ROLES.researcher }];
-      } else {
-        prev.roles = user.roles;
-      }
-      prev.additionalEmail = isNil(user.additionalEmail) ? '' : user.additionalEmail;
-      prev.institutionId = user.institutionId;
-      prev.institutionList = institutionList;
-      prev.profile.academicEmail = researcherProps.academicEmail || user.email;
-      prev.profile.address1 = researcherProps.address1;
-      prev.profile.address2 = researcherProps.address2;
-      prev.profile.checkNotifications = (researcherProps.checkNotifications === 'true');
-      prev.profile.city = researcherProps.city;
-      prev.profile.completed = researcherProps.completed;
-      prev.profile.country = researcherProps.country;
-      prev.profile.department = researcherProps.department;
-      prev.profile.division = researcherProps.division;
-      prev.profile.eRACommonsID = researcherProps.eraCommonsId;
-      prev.profile.havePI = researcherProps.havePI;
-      prev.profile.isThePI = researcherProps.isThePI;
-      prev.profile.linkedIn = researcherProps.linkedIn;
-      prev.profile.orcid = researcherProps.orcid;
-      prev.profile.piEmail = researcherProps.piEmail;
-      prev.profile.piName = researcherProps.piName;
-      prev.profile.profileName = user.displayName;
-      prev.profile.pubmedID = researcherProps.pubmedID;
-      prev.profile.researcherGate = researcherProps.researcherGate;
-      prev.profile.scientificURL = researcherProps.scientificURL;
-      prev.profile.state = researcherProps.state;
-      prev.profile.zipcode = researcherProps.zipcode;
-      return prev;
-    }, () => {
-      if (this.state.profile.completed !== undefined && this.state.profile.completed !== '') {
-        this.setState(prev => {
-          prev.profile.completed = JSON.parse(this.state.profile.completed);
-          return prev;
-        });
-      }
-    });
-  };
+    init();
+  }, [props.history]);
 
-  handleChange = (event) => {
-    let field = event.target.name;
-    let value = event.target.value;
+  useEffect(() => {
+    const isValid = (value) => {
+      let isValid = false;
+      if (trim(value.toString()) !== '' && value !== null && value !== undefined) {
```

Comment:
If value is null or undefined, this will throw an error with `value.toString()`. I suggest using Lodash `isNil` and `isEmpty` methods to simplify this.

Reply:
I misspoke here ... `trim` will handle nil and undefined just fine and return an empty string. Maybe wrap that in `isEmpty`?
```suggestion
      if (!isEmpty(trim(value.toString()))) {
```

Reply:
Fixed and simplified

Reply:
I'm still seeing the unfixed version, are your updates pushed to origin?